opte/src/engine/layer.rs

```diff
@@ -1228,6 +1239,16 @@ impl<'a> RuleTable {
         Err(Error::RuleNotFound { id })
     }
 
+    pub fn rule_id(&self, query_rule: &Rule<Finalized>) -> Option<RuleId> {
+        for (rule_id, rule) in self.rules.iter() {
```

Comment:
I usually reach for [find](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.find) for something like this, but it's not much clearer given how simple the predicate is.

Reply:
Yea, in this case it also needs to be followed with an `Option::map()`. In my Erlang days I was all about HOFs, but I think the last 6 years of C has made me imperative again. In this case it's small and reads fine, so I'm going to change it.

Also, the first time I read this too quickly and thought you were referring to the function name. So now that function name is also `find_rule`...which honestly is probably more clear.